### ISSUE TITLE
Go to next step after clicking element in highlighted area.

### ISSUE THREAD

**Issue by adamsoffer** [elrumordelaluz/reactour#201]
Hey, thanks for the awesome lib! Question - is it possible to navigate to another step by clicking somewhere in the highlighted area (not the next button)?

**Comment by adamsoffer** [comment]
Might be able to get away with observe? Doesn't seem to work though. Even if I observe a parent element...

**Comment by adamsoffer** [comment]
Alright figured out a workaround. If you're using next.js you can simply listen for a change in route inside `content`.

```
      content: ({ goTo }) => {
        Router.events.on('routeChangeComplete', url => {
          if (url == '/connect-wallet') goTo(1)
        })
        return (
          <div>
            <Styled.h2 sx={{ mb: 2 }}>Connect Wallet</Styled.h2>
            <Styled.p>First things first. Let's connect your wallet.</Styled.p>
          </div>
        )
      },
```

**Comment by adamsoffer** [comment]
@elrumordelaluz any ideas?

**Comment by georgiosgiatsidis** [comment]
Is anyone working on this?

**Comment by priyankatakkekar** [comment]
Is this feature available in version 1.16.1?

**Comment by dqvux** [comment]
Temporarily we can set an id for nextButton:
`
<Tour
          steps={steps}
          isOpen
          nextButton={
            <button id="_tour_next_btn">Next</button>
          }
        />
`
then use "action" in step:
`
const step = [{
        selector: "...",
        content: "...",
        action: (node) => {
          const nextBtn = document.getElementById("_tour_next_btn");
          if (nextBtn) nextBtn.style.display = "none"; // hide nextButton
          node.onclick = () => {  // handle clicking element in highlighted area
            setTimeout(() => {
              if (nextBtn) {
                  nextBtn.click(); // go to next step
                  nextBtn.style.display = "block"; // show nextButton
            }, 500); // maybe need a few time for other rendering
          };
        },
        stepInteraction: true,
      },
    ]
`

**Comment by dqvux** [comment]
Temporarily we can set an id for nextButton:
```js
<Tour 
    steps={steps}
    isOpen
    nextButton={<button id="_tour_next_btn">Next</button>}
/>
```

then use "action" in step:
```js
const steps = [
     {
        selector: "...",
        content: "...",
        action: (node) => {
          const nextBtn = document.getElementById("_tour_next_btn");
          if (nextBtn) nextBtn.style.display = "none"; // hide nextButton
          node.onclick = () => {  // handle clicking element in highlighted area
            setTimeout(() => {
              if (nextBtn) {
                  nextBtn.click(); // go to next step
                  nextBtn.style.display = "block"; // show nextButton
            }, 500); // maybe need a few time for other rendering
          };
        },
        stepInteraction: true,
      },
    ]
```

**Comment by JewNinja** [comment]
I resolved the problem in the following way:
https://jsfiddle.net/VanMortem/rez3L6o1/ (step 3)

also at the last step I had to create this:
https://jsfiddle.net/VanMortem/rez3L6o1/ (step 4)
https://jsfiddle.net/VanMortem/63hrcfwm/1/

**Comment by elrumordelaluz** [comment]
This is doable in `v2` through [onClickMask](https://github.com/elrumordelaluz/reactour/tree/master/packages/tour#onclickmask-clickprops-clickprops--void). [Here](https://reactour.vercel.app/#mask-click) is an example.
Closing but feel free to re-open or create a new one in any case.

**Comment by dreamfalcon** [comment]
> This is doable in `v2` through [onClickMask](https://github.com/elrumordelaluz/reactour/tree/master/packages/tour#onclickmask-clickprops-clickprops--void). [Here](https://reactour.vercel.app/#mask-click) is an example. Closing but feel free to re-open or create a new one in any case.

But this only works for clicking on the mask, not the highlighted area.
Is possible for certain steps to go next after clicking the highlighted element?

**Comment by elrumordelaluz** [comment]
> > This is doable in `v2` through [onClickMask](https://github.com/elrumordelaluz/reactour/tree/master/packages/tour#onclickmask-clickprops-clickprops--void). [Here](https://reactour.vercel.app/#mask-click) is an example. Closing but feel free to re-open or create a new one in any case.
> 
> But this only works for clicking on the mask, not the highlighted area. Is possible for certain steps to go next after clicking the highlighted element?

Did you tried using [onClickHighlighted](https://github.com/elrumordelaluz/reactour/tree/main/packages/tour#onclickhighlighted-mouseeventhandlersvgrectelement)

**Comment by dreamfalcon** [comment]
> > > This is doable in `v2` through [onClickMask](https://github.com/elrumordelaluz/reactour/tree/master/packages/tour#onclickmask-clickprops-clickprops--void). [Here](https://reactour.vercel.app/#mask-click) is an example. Closing but feel free to re-open or create a new one in any case.
> > 
> > 
> > But this only works for clicking on the mask, not the highlighted area. Is possible for certain steps to go next after clicking the highlighted element?
> 
> Did you tried using [onClickHighlighted](https://github.com/elrumordelaluz/reactour/tree/main/packages/tour#onclickhighlighted-mouseeventhandlersvgrectelement)

Only works with disableInteraction and the menu button stops working.

**Comment by elrumordelaluz** [comment]
if you want interaction, why don't use `setCurrentStep` from `useTour` hook instead?

**Comment by gnowland** [comment]
@elrumordelaluz  [onClickHighlighted](https://github.com/elrumordelaluz/reactour/tree/main/packages/tour#onclickhighlighted-mouseeventhandlersvgrectelement) doesn't have access to the methods `setIsOpen`, `setCurrentStep`, `setSteps`, or `setMeta`, so this is not an acceptable solution. We need to be able to interact with the hook when a user clicks within the highlighted area (not the outside mask). Thank you.

**Comment by elrumordelaluz** [comment]
@gnowland you were right, thank you! Now, [onClickHighlighted](https://github.com/elrumordelaluz/reactour/tree/main/packages/tour#onclickhighlighted-e-mouseeventhandlersvgrectelement-clickprops-clickprops--void) has access to all those methods. Should be available at `@reactour/tour@3.6.0`.

**Comment by gnowland** [comment]
This is fantastic news, thank you so much for your quick turnaround @elrumordelaluz!!!
